src/main/java/jenkins/plugins/git/GitCredentialBindings.java

```diff
@@ -0,0 +1,86 @@
+package jenkins.plugins.git;
+
+import com.cloudbees.plugins.credentials.common.StandardCredentials;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.EnvVars;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Executor;
+import hudson.model.Node;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.git.GitTool;
+import hudson.plugins.git.util.GitUtils;
+import org.jenkinsci.plugins.gitclient.GitClient;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Map;
+
+public interface GitCredentialBindings {
+
+    /**
+     * Sets secret or public pair value(s)
+     * @param credentials The credentials {@link com.cloudbees.plugins.credentials.common.StandardCredentials}. Cannot be null
+     * @param secretValues The values{@link java.util.Map} to be hidden in build logs
+     * @param publicValues The values{@link java.util.Map} to be visible in build logs
+     **/
+    void setCredentialPairBindings(@NonNull StandardCredentials credentials, Map<String,String> secretValues, Map<String,String> publicValues);
+
+    /**
+     * Set Git specific environment variable
+     * @param git GitClient {@link org.jenkinsci.plugins.gitclient.GitClient}. Cannot be null.
+     * @param secretValues The values{@link java.util.Map} to be hidden in build logs
+     * @param publicValues The values{@link java.util.Map} to be visible in build logs
+     **/
+    void setGitEnvironmentVariables(@NonNull GitClient git, Map<String,String> secretValues, Map<String,String> publicValues) throws IOException, InterruptedException;
+
+    /**
+     * Use GitClient to perform git operations in a repository. Using Git implementations JGit/JGit Apache/Cli Git
+     * @param gitExe The path {@link java.lang.String} to git executable {@link org.jenkinsci.plugins.gitclient.Git#using(String)}
+     * @param repository The path {@link java.lang.String} to working directory {@link org.jenkinsci.plugins.gitclient.Git#in(File)}
+     * @param env The environment values {@link hudson.EnvVars}
+     * @param listener The task listener.
+     * @return a GitClient implementation {@link org.jenkinsci.plugins.gitclient.GitClient}
+     **/
+    GitClient getGitClientInstance(String gitExe, FilePath repository,
```

Comment:
why is this added in the contract to be implemented? Can we not provide the logic assuming the implementation does not change for any reason?

Reply:
similar to getCliGitTool